jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/profiles/ProfileAssumeRoleProvider.kt

```diff
@@ -0,0 +1,80 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.credentials.profiles
+
+import software.amazon.awssdk.auth.credentials.AwsCredentials
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider
+import software.amazon.awssdk.profiles.Profile
+import software.amazon.awssdk.profiles.ProfileProperty
+import software.amazon.awssdk.services.sts.StsClient
+import software.amazon.awssdk.services.sts.auth.StsAssumeRoleCredentialsProvider
+import software.amazon.awssdk.services.sts.model.AssumeRoleRequest
+import software.amazon.awssdk.utils.SdkAutoCloseable
+import software.aws.toolkits.jetbrains.core.credentials.promptForMfaToken
+import java.util.function.Supplier
+
+class ProfileAssumeRoleProvider(private val stsClient: StsClient, private val parentProvider: AwsCredentialsProvider, profile: Profile) :
+    AwsCredentialsProvider, SdkAutoCloseable {
+    private val credentialsProvider: StsAssumeRoleCredentialsProvider
+
+    init {
+        val roleArn = profile.requiredProperty(ProfileProperty.ROLE_ARN)
+        val roleSessionName = profile.property(ProfileProperty.ROLE_SESSION_NAME)
+            .orElseGet { "aws-toolkit-jetbrains-${System.currentTimeMillis()}" }
+        val externalId = profile.property(ProfileProperty.EXTERNAL_ID)
+            .orElse(null)
```

Comment:
why the aggressive new-lining? this should be well within the 140 chars no?

Reply:
You are correct, it was copy pasted as is, but they fit now